lib/Core/Memory.cpp

```diff
@@ -527,7 +587,11 @@ void ObjectState::write(unsigned offset, const ref<Expr> &value) {
     if (isConcreteOnly() && Context::get().isLittleEndian()) {
       // Extend bit and write as single byte
       w = w > 1 ? w : Expr::Int8;
+#if LLVM_VERSION_CODE < LLVM_VERSION(9, 0)
```

Comment:
I think it would be actually better to just not do this optimisation for older LLVM versions. With the speed llvm is getting dropped, I don't think it is going to be an issue for long.  Also it's a great "soft" push for later LLVM: "if you use KLEE with llvm 9 you will see faster interpretation", who is not going to want to upgrade.

Reply:
Well, here it would be good to be able to compare performance with different versions.

Reply:
What's the point of that. If newer ones are faster we need to move forward as usual. If they are not, we can't be stuck in the past so also move forward.

I was just considering simplifying the code, since this isn't a functionality adding patch, so it's less important to support llvm versions.

Btw would it be really bad to put your implementation of these functions in the `llvm` namespace?

Reply:
Like the commit message of this part says, it is exactly the same code as LLVM already provides. The only issue is that it was not accessible via a public API.

Reply:
Can't you just declare it? If those symbols are public, it should be fine. I kind of just want to avoid this `ifdef`, which I think you could by a bit of name hacking.

Reply:
Nope. This way, it's cleaner.

Reply:
check my previous comment which contains the problem to a localized place.